src/main/java/edu/wpi/grip/ui/SocketControlView.java

```diff
@@ -0,0 +1,172 @@
+
+        // Set an input control of some sort based on the "view" field of the socket hint
+        switch (socketHint.getView()) {
+            case NONE:
+                // The hint says not to show a control, so there's nothing to do here.
+                break;
+
+            case SPINNER: {
+                // To use a spinner, the socket should be able to hold a number
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType()));
+
+                SpinnerValueFactory spinnerValueFactory;
+                Number initialValue = (Number) socket.getValue();
+
+                Number min = Double.MIN_VALUE, max = Double.MAX_VALUE;
+                if (domain != null && domain.length == 2) {
+                    // A spinner can have a min and max, if the domain holds two values.
+                    min = (Number) domain[0];
+                    max = (Number) domain[1];
+                }
+
+                spinnerValueFactory = new SpinnerValueFactory.DoubleSpinnerValueFactory(
+                        min.doubleValue(), max.doubleValue(), initialValue.doubleValue());
+
+                Spinner<Double> spinner = new Spinner<>(spinnerValueFactory);
+                spinner.setEditable(true);
+
+                this.controlPane.getChildren().add(spinner);
+                this.valueProperty.bindBidirectional(spinnerValueFactory.valueProperty());
+
+                break;
+            }
+            case SLIDER: {
+                // To use a slider, the socket should be able to hold a number
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType()));
+
+                // A slider needs a min and max, so the domain should hold two values.
+                checkArgument(domain != null && domain.length == 2);
+                double min = ((Number) domain[0]).doubleValue(),
+                        max = ((Number) domain[1]).doubleValue(),
+                        initialValue = ((Number) socket.getValue()).doubleValue();
+
+                Slider slider = new Slider(min, max, initialValue);
+                this.controlPane.getChildren().add(slider);
+                this.valueProperty.bindBidirectional(slider.valueProperty());
+
+                break;
+            }
+            case RANGE: {
+                // A range produces an array of two numbers, so the socket should expect an array that can hold numbers
+                checkArgument(socketHint.getType().isArray());
+                checkArgument(Number.class.isAssignableFrom(socketHint.getType().getComponentType()));
+
+                // Same
+                checkArgument(domain != null && domain.length == 2);
+                double min = ((Number) domain[0]).doubleValue(),
+                        max = ((Number) domain[1]).doubleValue(),
+                        initialValue = ((Number) socket.getValue()).doubleValue();
+
+                RangeSlider rangeSlider = new RangeSlider(min, max, initialValue, initialValue + 10);
```

Comment:
Why is the initial value defined with 10 added to it?
Add a comment explaining this.